_includes/exhibits.js

```diff
@@ -0,0 +1,1929 @@
+    if (this.edit) {
+      return this.state.editing;
+    }
+    return 0;
+  },
+  set editing(_e) {
+    const e = parseInt(_e, 10);
+    this.state.editing = modulo(e, 3);
+    this.newView(true);
+  },
+
+  get drawing() {
+    return this.state.drawing;
+  },
+  set drawing(_d) {
+    const d = parseInt(_d, 10);
+    this.state.drawing = modulo(d, 3);
+    this.newView(true);
+  },
+
+  get mouseXY() {
+    const e = this.state.mouseEvent;
+    const pos = OpenSeadragon.getMousePosition(e);
+    return this.normalize(pos);
+  },
+  set mouseXY(e) {
+    this.state.mouseEvent = e;
+  },
+
+  /*
+   * Hash Keys
+   */
+
+  get v() {
+    return this.state.v;
+  },
+  set v(_v) {
+    const viewer = this.viewer;
+    this.state.v = _v.map(parseFloat);
+  },
+
+  get g() {
+    const g = this.state.g;
+    const count = this.cgs.length;
+    return g < count ? g : 0;
+  },
+  set g(_g) {
+    const g = parseInt(_g, 10);
+    const count = this.cgs.length;
+    this.state.g = modulo(g, count);
+  },
+
+  /*
+   * Exhibit Hash Keys
+   */
+
+  get w() {
+    const w = this.state.w[this.s] || 0;
+    const count = this.waypoints.length;
+    return w < count ? w : 0;
+  },
+
+  set w(_w) {
+    const w = parseInt(_w, 10);
+    const count = this.waypoints.length;
+    this.state.w[this.s] = modulo(w, count);
+
+    // Set group, viewport from waypoint
+    const waypoint = this.waypoint;
+
+    this.slower();
+    this.g = gFromWaypoint(waypoint, this.cgs);
+    this.v = vFromWaypoint(waypoint);
+    this.o = oFromWaypoint(waypoint);
+  },
+
+  get s() {
+    const s = this.state.s;
+    const count = this.stories.length;
+    return s < count ? s : 0;
+  },
+  set s(_s) {
+    const s = parseInt(_s, 10);
+    const count = this.stories.length;
+    this.state.s = modulo(s, count);
+
+    // Update waypoint
+    this.w = this.w;
+    if (this.s != s || !this.editing) {
+      const waypoint = this.waypoint;
+      this.d = dFromWaypoint(waypoint);
+      this.n = nFromWaypoint(waypoint);
+    }
+  },
+
+  /*
+   * Tag Hash Keys
+   */
+
+  get o() {
+    return this.state.o;
+  },
+  set o(_o) {
+    this.state.o = _o.map(parseFloat);
+  },
+
+  get d() {
+    return this.state.description;
+  },
+  set d(_d) {
+    this.state.description = '' + _d;
+  },
+
+  get n() {
+    return this.state.name;
+  },
+  set n(_n) {
+    this.state.name = '' + _n;
+  },
+
+  /*
+   * Configuration State
+   */
+  get changed() {
+    return this.state.changed;
+  },
+  set changed(_c) {
+    this.state.changed = !!_c;
+  },
+
+  get yaml() {
+    const config = {
+      'Exhibit': {
+         'Stories': this.stories,
+         'Channels': this.chans,
+         'Layout': this.layout,
+         'Images': this.images,
+         'Groups': this.cgs,
+      }
+    };
+    return jsyaml.safeDump(config, {
+      lineWidth: 40,
+      sortKeys: sort_keys,
+      noCompatMode: true,
+    });
+  },
+
+  get design() {
+    return deepCopy(this.state.design);
```

Comment:
This prevents modification of the design state directly. All the setters that affect the design ensure that the design change is recorded in "state.change" whenever the design changes.